src/serializer/ResidualHeapSerializer.js

```diff
@@ -1101,6 +1101,13 @@ export class ResidualHeapSerializer {
                   mightHaveBeenDeleted
                 );
               });
+
+              // Although the property needs to be delayed, we still want to emit "undefined"
+              // as part of the object literal to ensure a consistent property ordering.
+              if (!mightHaveBeenDeleted) {
```

Comment:
I'm not really sure, that `!mightHaveBeenDeleted` is the right condition here to check against.
But without this check tests were failing with the following error:

```
test/serializer/abstract/PutValue11.js
Output mismatch!
original code
var c = global.__abstract ? __abstract("boolean", "false") : false;
a = {};
if (c) a.f = a;

inspect = function() { return 'f' in a; }

output of inspect() on original code
false
generated code in iteration 0
var c;
(function () {
  var __empty = {};

  var _$0_unique27277 = this;

  function _3_unique27277() {
    return 'f' in _$0_unique27277.a;
  }

  c = void 0;
  var _0_unique27277 = false;
  c = _0_unique27277;
  var _1_unique27277 = {
    f: void 0
  };

  var _2_unique27277 = _0_unique27277 ? _1_unique27277 : __empty;

  if (_2_unique27277 !== __empty) _1_unique27277.f = _2_unique27277;
  a = _1_unique27277;
  inspect = _3_unique27277;
}).call(this);
output of inspect() on last generated code iteration
true
```